spec/puppetserver/utils/http_utilities_spec.rb

```diff
@@ -0,0 +1,60 @@
+require 'spec_helper'
+require 'puppetserver/utils/http_utilities'
+require 'puppetserver/ca/logger'
+require 'puppetserver/ca/generate_action'
+require 'utils/ssl'
+
+RSpec.describe Puppetserver::Utils::HttpUtilities do
+  include Utils::SSL
+
+  subject { Puppetserver::Utils::HttpUtilities }
+
+  it 'creates the frickin store correctly' do
+    stdout = StringIO.new
+    stderr = StringIO.new
+    logger = Puppetserver::Ca::Logger.new(:info, stdout, stderr)
+    generate_action = Puppetserver::Ca::GenerateAction.new(logger)
+
+    Dir.mktmpdir do |tmpdir|
+      cadir = tmpdir
+      cacert = File.join(tmpdir, 'ca_crt.pem')
+      cakey = File.join(tmpdir, 'ca_key.pem')
+      rootkey = File.join(tmpdir, 'root_key.pem')
+      cacrl = File.join(tmpdir, 'ca_crl.pem')
+
+
+      settings = {
+        ca_ttl: (5 * 365 * 24 * 60 * 60),
+        keylength: 2048,
+        root_ca_name: "root",
+        ca_name: 'leaf',
+        cadir: cadir,
+        cacert: cacert,
+        cakey: cakey,
+        rootkey: rootkey,
+        cacrl: cacrl
+      }
+
+      digest = generate_action.default_signing_digest
+      generate_action.generate_root_and_intermediate_ca(settings, digest)
```

Comment:
I think we should probably use this here in testing rather than the ssl utils (I still wasn't able to get it to fail with either of these methods...) But that GenerateAction will eventually become our CA construct?